lib/posix-process/signal/rt_tgsigqueueinfo.c

```diff
@@ -0,0 +1,45 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2023, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+#include <errno.h>
+#include <signal.h>
+
+#include <uk/essentials.h>
+#include <uk/process.h>
+#include <uk/syscall.h>
+
+#include "signal.h"
+
+UK_LLSYSCALL_R_DEFINE(int, rt_tgsigqueueinfo,
+		      int, tgid, int, tid,
+		      int, signum, siginfo_t *, siginfo)
+{
+	struct posix_process *proc;
+	struct posix_thread *pthread;
+
+	if (unlikely(!IS_VALID(signum)))
+		return -EINVAL;
+
+	if (unlikely(!siginfo))
```

Comment:
Interesting how the `EFAULT` errno is not specified in the manuals but Linux also returns this in case of `copy_from_user` failure. :thinking:

Reply:
They do mention it in the man pages of some of the syscalls, e.g. sigprocmask(2) states that:
> EFAULT The set or oldset argument points outside the process's allocated address space.

If we wanted to support this we would need to assign a different allocator per process. Which would also allow us among others to limit per-process memory consumption, yet I'm not sure if that's something we want.